hil/api.py

```diff
@@ -445,6 +454,15 @@ def node_detach_network(node, nic, network):
         .filter_by(nic=nic, network=network).first()
     if attachment is None:
         raise BadArgumentError("The network is not attached to the nic.")
+    # Check if any other VLANs are attached before removing the native VLAN
+    if attachment.channel == 'vlan/native' \
+        and db.session.query(model.NetworkAttachment) \
+            .filter(model.NetworkAttachment.channel != 'vlan/native') \
+            .filter(model.NetworkAttachment.nic_id == nic.id) \
+            .count() > 0:
+        raise BlockedError(
+                "Please remove all trunked VLANs "
+                "before removing the native VLAN.")
```

Comment:
Having talked about this at the meeting, I think we're going to go with the approach outlined in #854, so I'm going to cut my review short. @Izhmash, sorry I didn't think of this before you went ahead and did all this work...